src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -159,108 +161,105 @@ public RestoreCommand()
             {
                 // Restore a project.json file using the directory as the Id
                 Console.LogVerbose($"Reading project file {Arguments[0]}");
-                var projectDirectory = Path.GetDirectoryName(inputPath);
 
                 projectJsonPath = inputPath;
-
-                packageSpec = JsonPackageSpecReader.GetPackageSpec(
-                    File.ReadAllText(projectJsonPath),
-                    Path.GetFileName(projectDirectory),
-                    projectJsonPath);
+                projectName = Path.GetFileName(projectDirectory);
             }
-            else if (MsBuildUtility.IsMsBuildBasedProject(inputPath))
+            else if (ProjectHelper.UnsupportedProjectExtensions.Contains(Path.GetExtension(inputPath)))
```

Comment:
This is still a bit hacky. We should also deal with project.json where there is no .csproj or any file next to it.